src/service/viewer-impl.js

```diff
@@ -844,7 +853,7 @@ export class Viewer {
     if (!this.hasCapability('fragment')) {
       return Promise.resolve('');
     }
-    return this.sendMessageUnreliable_('fragment', undefined, true).then(
+    return this.sendMessageCancelUnsent('fragment', undefined, true).then(
```

Comment:
There will be only one `getFragment` call.
Without the modified queued up change in this PR, the previous `sendMessageUnreliable_` is wrong, since it will get a resolved promise without getting the fragment if the message is sent before message channel is established. But now I think we should use `sendMessageCancelUnsent` because it now only resolves when the message channel is established and get a response from the viewer, which is exactly what we want.